common/workunit/workunit.cpp

```diff
@@ -1888,6 +1892,21 @@ class CConstWUArrayIterator : public CInterface, implements IConstWorkUnitIterat
     }
     IConstWorkUnitInfo & query() { return *cur; }
 };
+
+class CConstWUFieldUsageFileIterator : public CInterface, implements IConstWUFieldUsageFileIterator
+{
+public:
+   IMPLEMENT_IINTERFACE;
+   CConstWUFieldUsageFileIterator(IPropertyTreeIterator * tree) { iter.setown(tree); }
+   bool                 first() { item.clear(); return iter->first(); }
+   bool                 isValid() { return iter->isValid(); }
+   bool                 next() { item.clear(); return iter->next(); }
+   IConstWUFieldUsageFile & query() { if (!item) item.setown(new CLocalWUFieldUsageFile(iter->get())); return *item; };
```

Comment:
Would it make sense to create the CLocalWUFieldUsageFile instance in the CTOR instead of here?  Then the query method wont have to make the test with every invocation